File: fizbuzz.rb
#!/usr/bin/env ruby

def fizzbuzz(n)
  return "FizzBuzz" if n % 15 == 0
  return "Fizz" if n % 3 == 0
  return "Buzz" if n % 5 == 0
  return n.to_s
end

def run_fizzbuzz(count)
  (1..count).each do |i|
    puts fizzbuzz(i)
  end
end

# コマンドライン引数を処理
count = 100 # デフォルト値
if ARGV.length > 0
  begin
    count = Integer(ARGV[0])
    if count <= 0
      puts "Please provide a positive number of iterations"
      exit(1)
    end
  rescue ArgumentError
    puts "Please provide a valid number"
    exit(1)
  end
end

run_fizzbuzz(count)

File: fizbuzz.c
#include <stdio.h>
#include <stdlib.h>

/**
 * FizzBuzz implementation in C with command-line arguments support
 */
int main(int argc, char *argv[]) {
    int count = 100; // デフォルトは100回
    
    // コマンドライン引数を処理
    if (argc > 1) {
        count = atoi(argv[1]);
        if (count <= 0) {
            printf("Please provide a positive number of iterations\n");
            return 1;
        }
    }
    
    // FizzBuzzシーケンスを出力
    for (int i = 1; i <= count; i++) {
        if (i % 3 == 0 && i % 5 == 0) {
            printf("FizzBuzz\n");
        } else if (i % 3 == 0) {
            printf("Fizz\n");
        } else if (i % 5 == 0) {
            printf("Buzz\n");
        } else {
            printf("%d\n", i);
        }
    }
    return 0;
}